### ISSUE TITLE
Build image should be able to overwrite

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#49]
**Reported by brechin on 2 Jul 2002 05:58 UTC**
```
When users click to build the client image, they should
be given the option to overwrite the image (i.e. delete
the existing image and create the new one).  This would
eliminate many headaches for people and would make the
problem of existing client images more manageable.
```

**Comment by dikim33** [comment]
**Comment by richferri on 2 Jul 2002 12:36 UTC**
```

Personally, I think it's more clear to the end user to have
the ability to delete an image, and then create a new one.
The concept of 'overwriting' connotes space reclaimation and
management -- what happens when the new image is smaller
than the old? Does the 'extra' space at the old image get
grabage collected? And what happens when the new image is
larger then the old -- does new space get allocated from the
general pool of disk space? I think simply deleting an image
that is no longer needed, and then creating a new one is
very clear to the end user on how space is managed.
```

**Comment by dikim33** [comment]
**Comment by jenos on 3 Jul 2002 04:08 UTC**
```

So does the "overwrite" step imply a "Delete, Create" 
process?  If not, I see Rich's point.  If so, having an 
overwrite option is an added measure of convenience.  
Space reclaimation connotations would be valid if it was 
a "delete, create" process, right?  Smaller or larger 
wouldn't matter?  Unless I'm misunderstanding which is 
entirely posssible.  ;-)
```

**Comment by dikim33** [comment]
**Comment by brechin on 3 Jul 2002 05:24 UTC**
```

Yes... as I said... "delete the existing image and create
the new one" would be my suggestion.  This eliminates the
need to worry about larger/smaller images.
```

**Comment by dikim33** [comment]
**Comment by dnl on 27 Aug 2005 17:04 UTC**
```

This was fixed for 4.2 when the ability to cancel an image
build was added.
```
